### PR TITLE
The ID of patients, notes and annotations can now be defined by the user

### PR DESCRIPTION
**Pull request by tschaffter**
Addresses the schema definition-site of https://github.com/nlpsandbox/data-node/blob/develop/server/openapi_server/dbmodels/annotation.py

Previously the ID of patients, notes and annotations was generated by the database. Because the ID is used to build the name of a resource, which is usually used to reference a resource, and because the ID was automatically generated, it was not possible transfer the content of a dataset to another Data Node instance (e.g. following the crash of a database). If we allow the user to specify the ID of all the resources created, this makes it possible to copy/move the content of a Data Node from one instance to another.

### CODE REVIEW

File: openapi/commons/components/parameters/AnnotationName.yaml [hunk 1]
```diff
@@ -1,3 +1,3 @@
-description: Resource name of the annotation record, of the form datasets/{datasetId}/annotationStores/{annotationStoreId}/annotations/{annotationId}
+description: The resource name of the annotation record, of the form datasets/{datasetId}/annotationStores/{annotationStoreId}/annotations/{annotationId}
```
Comment (thomasyu888) [inline openapi/commons/components/parameters/AnnotationName.yaml:1]: Not as important, but can add a pattern to this:
```
pattern: ^datasets\/[a-zA-Z0-9]+\/annotationStores\/[a-zA-Z0-9]+\/annotations\/[a-zA-Z0-9]+$
```

File: openapi/commons/components/schemas/Note.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ properties:
     type: string
     example: "loinc:LP29684-5"
   patientId:
-    $ref: ../parameters/PatientResourceId.yaml
+    $ref: ../parameters/PatientId.yaml
 required:
   - text
```
Comment (thomasyu888) [inline openapi/commons/components/schemas/Note.yaml:17]: Should `patientId` and `id` be required?

File: openapi/data-node/components/schemas/NoteCreateRequest.yaml [hunk 3]
```diff
@@ -4,13 +4,13 @@ properties:
...
+    example: "On 12/26/2020, Ms. Chloe Price met with Dr. Prescott in Seattle."
   noteType:
     description: The note type (LOINC concept)
     type: string
     example: "loinc:LP29684-5"
   patientId:
-    $ref: ../../../commons/components/parameters/PatientResourceId.yaml
+    $ref: ../../../commons/components/parameters/PatientId.yaml
 required:
   - text
```
Comment (thomasyu888) [inline openapi/data-node/components/schemas/NoteCreateRequest.yaml:15]: Add patientId to required

File: openapi/commons/components/schemas/Note.yaml (hunk 2 above)
Comment (tschaffter) [reply openapi/commons/components/schemas/Note.yaml:17]: OK

File: openapi/data-node/components/schemas/NoteCreateRequest.yaml (hunk 3 above)
Comment (tschaffter) [reply openapi/data-node/components/schemas/NoteCreateRequest.yaml:15]: Done

File: openapi/commons/components/parameters/AnnotationName.yaml (hunk 1 above)
Comment (tschaffter) [reply openapi/commons/components/parameters/AnnotationName.yaml:1]: Done

**Comment by tschaffter** [comment]
@thomasyu888 I implemented your suggestions. I'll work on the Data Node implementation and close this ticket once the implementation is successful.

File: openapi/commons/components/parameters/AnnotationName.yaml (hunk 1 above)
Comment (tschaffter) [reply openapi/commons/components/parameters/AnnotationName.yaml:1]: I reused the pattern for IDs

**Pull request merged**